File: backend-go/internal/model/ai_news.go
package model

import (
	"database/sql/driver"
	"encoding/json"
	"fmt"
	"time"
)

// StringArray 自定义字符串数组类型，兼容 MySQL JSON 字段
type StringArray []string

func (a StringArray) Value() (driver.Value, error) {
	if a == nil {
		return "[]", nil
	}
	return json.Marshal(a)
}

func (a *StringArray) Scan(value interface{}) error {
	if value == nil {
		*a = []string{}
		return nil
	}
	bytes, ok := value.([]byte)
	if !ok {
		return fmt.Errorf("StringArray.Scan: 无法转换 %T 到 []byte", value)
	}
	return json.Unmarshal(bytes, a)
}

// AiNews AI 过滤新闻模型
type AiNews struct {
	ID            string      `gorm:"type:char(36);primaryKey" json:"id"`
	Source        string      `gorm:"type:varchar(64)" json:"source"`
	SourceURL     *string     `gorm:"type:varchar(1024)" json:"source_url,omitempty"`
	Title         string      `gorm:"type:varchar(512)" json:"title"`
	Summary       *string     `gorm:"type:text" json:"summary,omitempty"`
	Content       *string     `gorm:"type:text" json:"content,omitempty"`
	RelatedFunds  StringArray `gorm:"type:json;default:'[]'" json:"related_funds"`
	Sentiment     *int16      `json:"sentiment,omitempty"`
	Importance    *int16      `json:"importance,omitempty"`
	Tags          StringArray `gorm:"type:json;default:'[]'" json:"tags"`
	PublishedAt   *time.Time  `json:"published_at,omitempty"`
	ProcessedAt   time.Time   `json:"processed_at"`
	CreatedAt     time.Time   `json:"created_at"`
}

func (AiNews) TableName() string { return "ai_news" }

File: ai-service-python/app/models/ai_news.py
from datetime import datetime

from sqlalchemy import String, SmallInteger, Text, DateTime
from sqlalchemy.dialects.postgresql import UUID, ARRAY
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, UUIDMixin


class AiNews(UUIDMixin, Base):
    __tablename__ = "ai_news"

    source: Mapped[str] = mapped_column(String(64), nullable=False)
    source_url: Mapped[str | None] = mapped_column(String(1024))
    title: Mapped[str] = mapped_column(String(512), nullable=False)
    summary: Mapped[str | None] = mapped_column(Text)
    content: Mapped[str | None] = mapped_column(Text)
    related_funds: Mapped[list[str]] = mapped_column(ARRAY(UUID(as_uuid=True)), default=list)
    sentiment: Mapped[int | None] = mapped_column(SmallInteger)
    importance: Mapped[int | None] = mapped_column(SmallInteger)
    tags: Mapped[list[str]] = mapped_column(ARRAY(String(64)), default=list)
    published_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    processed_at: Mapped[datetime] = mapped_column(DateTime(timezone=True))
